stdlib/public/core/Algorithm.swift

```diff
@@ -177,9 +177,8 @@ public struct EnumerateGenerator<
   ///
   /// - Requires: No preceding call to `self.next()` has returned `nil`.
   public mutating func next() -> Element? {
-    let b = base.next()
```

Comment:
This could be implemented using `Optional.map`:
```swift
public mutating func next() -> Element? {
  return base.next().map { (index: count++, element: $0) }
}
```